app/models/in_progress_form.rb

```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: true
 class InProgressForm < ActiveRecord::Base
   attr_encrypted :form_data, key: Settings.db_encryption_key
+  validates(:form_data, presence: true)
+  before_save :serialize_form_data
 
   def self.form_for_user(form_id, user)
     InProgressForm.find_by(form_id: form_id, user_uuid: user.uuid)
   end
+
+  private
+
+  def serialize_form_data
+    self.form_data = form_data.to_json unless form_data.is_a?(String)
```

Comment:
@lihanli yep, plan is to have metadata be separate, but if you store the form_data as a json string before save but then merge in the metadata on render I think you'll get escape chars back in?

``` javascript
{
  "data":"{\"chapter1606\":true,\"veteranFullName\":{\"first\":\"John\",\"last\":\"Smith\"}",
  "meta":{"version":"1.2","returnUrl":"edu_benefits/page/3"}}
```